src/ui/public/directives/input_base64.js

```diff
@@ -0,0 +1,81 @@
+import { uiModules } from 'ui/modules';
+import { Observable } from 'rxjs/Rx';
+const module = uiModules.get('kibana');
+
+const $readFileContents = (file) => {
```

Comment:
Personal option, `createFileContent$` (create file content stream) reads a little better, especially when you're assigning it to a variable:

```js
const fileContent$ = createFileContent$(file)
```

Reply:
Makes sense, I will make it so.